prime_divisors/primes.rb

```diff
@@ -1,3 +1,17 @@
+# This program prints the value of "pi(n)", i.e. the numbers of primes
+# less than @p n.
+#
+# Arguments:
+#
+#     n: The number less than equal to which primes must be computed.
+#
+# Result:
+#
+#     n pi(n)
+#
+
+# @return The number of prime divisors of @p n.
+
 def n_prime_divisors(n)
```

Comment:
Yah, the 'compute pi only' seems best. Just need to be sure that every
program is doing the exact same thing (or as close as
humanly-taught-computer-possible).

On Wed, Jan 21, 2015 at 9:16 AM, Manav <notifications@github.com> wrote:

> In prime_divisors/primes.rb
> <https://github.com/zbard/benchmark/pull/3#discussion_r23276712>:
>
> > @@ -1,3 +1,17 @@
> > +# This program prints the value of "pi(n)", i.e. the numbers of primes
> > +# less than @p n.
> > +#
> > +# Arguments:
> > +#
> > +#     n: The number less than equal to which primes must be computed.
> > +#
> > +# Result:
> > +#
> > +#     n pi(n)
> > +#
> > +
> > +# @return The number of prime divisors of @p n.
> > +
> >  def n_prime_divisors(n)
>
> Ahh I get what you mean now. Yup, so either we can fix this, or skip these
> abstractions and compute pi only. What say?
> … <#14b0a99acd8b2c6e_>
> On Tuesday, January 20, 2015, Akshat Kumar <notifications@github.com>
> wrote: In prime_divisors/primes.rb <
> https://github.com/zbard/benchmark/pull/3#discussion-diff-23231587>: > @@
> -1,3 +1,17 @@ > +# This program prints the value of "pi(n)", i.e. the
> numbers of primes > +# less than @p <https://github.com/p> n. > +# > +#
> Arguments: > +# > +# n: The number less than equal to which primes must be
> computed. > +# > +# Result: > +# > +# n pi(n) > +# > + > +# @return
> <https://github.com/return> The number of prime divisors of @p
> <https://github.com/p> n. > + > def n_prime_divisors(n) irb>
> n_prime_divisors(4) => 3 irb> n_prime_divisors(8) => 3 irb>
> n_prime_divisors(14) => 3 … <#14b081275abfdf96_> On Tue, Jan 20, 2015 at
> 6:19 PM, Manav <notifications@github.com <javascript:_e(%7B%7D,'cvml','
> notifications@github.com');>> wrote: In prime_divisors/primes.rb <
> https://github.com/zbard/benchmark/pull/3#discussion-diff-23219544>: > @@
> -1,3 +1,17 @@ > +# This program prints the value of "pi(n)", i.e. the
> numbers of primes > +# less than @p <https://github.com/p> <
> https://github.com/p> n. > +# > +# Arguments: > +# > +# n: The number
> less than equal to which primes must be computed. > +# > +# Result: > +# >
> +# n pi(n) > +# > + > +# @return <https://github.com/return> <
> https://github.com/return> The number of prime divisors of @p
> <https://github.com/p> <https://github.com/p> n. > + > def
> n_prime_divisors(n) ? irb> n_prime_divisors(4) => 3 — Reply to this email
> directly or view it on GitHub <
> https://github.com/zbard/benchmark/pull/3/files#r23219544>. — Reply to
> this email directly or view it on GitHub <
> https://github.com/zbard/benchmark/pull/3/files#r23231587>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zbard/benchmark/pull/3/files#r23276712>.
>